### PR TITLE
Add lock_script_id and tx_hash

### PR DESCRIPTION
**Pull request by duanyytop**
- Insert `tx_hash` and `lock_script_id` to the withdrawal table block by block
- Add `tx_hash` and `lock_script_id` to the withdrawal history data when the `syncer` starts

### CODE REVIEW

File: internal/data/withdraw_extra_info.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+func (rp withdrawExtraInfoRepo) CreateExtraInfo(ctx context.Context, outPoint string, txHash string, lockScriptId uint) error {
+	outPointCrc := crc32.ChecksumIEEE([]byte(outPoint))
+	if err := rp.data.db.WithContext(ctx).Model(WithdrawCotaNftKvPair{}).Where("out_point_crc = ? AND out_point = ?", outPointCrc, outPoint).Updates(WithdrawCotaNftKvPair{TxHash: txHash, LockScriptId: lockScriptId}).Error; err != nil {
+		return err
+	}
+	return nil
+}
+
+func (rp withdrawExtraInfoRepo) FindAllQueryInfos(ctx context.Context) ([]biz.WithdrawQueryInfo, error) {
```
Comment (shaojunda) [inline internal/data/withdraw_extra_info.go:35]: It is recommended to pass `page` and `page_size to the function, and return data in batches.

File: internal/data/mint_cota_kv_pair.go [hunk 2]
```diff
@@ -74,32 +69,35 @@ func generateDefineWithdrawV1KvPairs(blockNumber uint64, entry biz.Entry, rp min
...
 			LockHashCrc:          lockHashCRC32,
+			LockScriptId:         senderLock.ID,
 			Version:              entry.Version,
 		})
 	}
 	return
 }
 
-func generateDefineWithdrawV0KvPairs(blockNumber uint64, entry biz.Entry, rp mintCotaKvPairRepo) (updatedDefineCotas []biz.DefineCotaNftKvPair, withdrawCotas []biz.WithdrawCotaNftKvPair, err error) {
+func generateMintV0KvPairs(blockNumber uint64, entry biz.Entry, rp mintCotaKvPairRepo) (defineCotas []biz.DefineCotaNftKvPair, withdrawCotas []biz.WithdrawCotaNftKvPair, err error) {
```
Comment (shaojunda) [inline internal/data/mint_cota_kv_pair.go:86]: Why change the name? this function is returning `DefineCotaNftKvPair ` and `WithdrawCotaNftKvPair `

File: internal/data/script_utils.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"encoding/hex"
+	"hash/crc32"
+
+	"github.com/nervina-labs/cota-nft-entries-syncer/internal/biz"
+	"github.com/nervina-labs/cota-nft-entries-syncer/internal/data/blockchain"
+)
+
+func GenerateSenderLock(entry biz.Entry) (lockScript biz.Script, lockHashStr string, lockHashCRC32 uint32, err error) {
```
Comment (shaojunda) [inline internal/data/script_utils.go:11]: It is recommended that this method only returns `LockScript`, put `lockHash` and `CRC` to other function

File: internal/service/withdraw_extra_info.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	var block *ckbTypes.Block
+	var lock biz.Script
+	for _, v := range queryInfos {
+		block, err = s.client.Rpc.GetBlockByNumber(ctx, v.BlockNumber)
+		if err != nil {
+			return err
+		}
+		for _, tx := range block.Transactions {
+			for _, input := range tx.Inputs {
+				if hex.EncodeToString(input.PreviousOutput.TxHash[12:]) == v.OutPoint[:40] {
```
Comment (shaojunda) [inline internal/service/withdraw_extra_info.go:44]: `hex.EncodeToString(input.PreviousOutput.TxHash[12:]) == v.OutPoint[:40]`  It is recommended to extract it into a method.

File: internal/service/withdraw_extra_info.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+					for _, output := range tx.Outputs {
+						lockHash, err := output.Lock.Hash()
```
Comment (shaojunda) [inline internal/service/withdraw_extra_info.go:46]: put the process of processing output into a method

File: internal/data/mint_cota_kv_pair.go (hunk 2 above)
Comment (duanyytop) [reply internal/data/mint_cota_kv_pair.go:86]: Mint is an action that will update `define` and `withdraw` SMT leaves. When we say minting CoTA NFTs,  the `DefineCotaNftKvPair` and `WithdrawCotaNftKvPair` must be updated. 

So I think `generateMintV0KvPairs ` is more simple and not less clear than `generateDefineWithdrawV0KvPairs`.

File: internal/service/withdraw_extra_info.go (hunk 5 above)
Comment (duanyytop) [reply internal/service/withdraw_extra_info.go:46]: Gotcha

File: internal/data/mint_cota_kv_pair.go (hunk 2 above)
Comment (shaojunda) [reply internal/data/mint_cota_kv_pair.go:86]: ok, It's better to use the `action` to name other functions.

File: internal/data/script_utils.go (hunk 3 above)
Comment (shaojunda) [reply internal/data/script_utils.go:11]: 1. follow the single responsibility principle

Comment (duanyytop) [reply internal/data/script_utils.go:11]: OK, maybe your idea is better.

File: internal/data/mint_cota_kv_pair.go (hunk 2 above)
Comment (duanyytop) [reply internal/data/mint_cota_kv_pair.go:86]: Now the other functions are all already actions, such as `transfer/withdraw/define/claim`.

File: internal/data/script_utils.go (hunk 3 above)
Comment (duanyytop) [reply internal/data/script_utils.go]: https://github.com/nervina-labs/cota-nft-entries-syncer/pull/14/commits/199e99a667a50c1859bc11d943d84bef845d7c40 cc @shaojunda

**Comment by duanyytop** [comment]
https://github.com/Magickbase/ckb-explorer-public-issues/issues/32 needs the NFT transactions with the `syncer` and the `aggregator` cc @shaojunda

**Review by shaojunda — APPROVED** [review]

**Pull request merged**